File: sample/src/main/java/com/leibo/repertory/widget/CustomViewActivity.java
package com.leibo.repertory.widget;

import android.view.View;

import com.leibo.baselib.base.BaseActivity;
import com.leibo.repertory.R;
import com.leibo.repertory.widget.banner.BannerViewActivity;
import com.leibo.repertory.widget.falling.FallingViewActivity;

/**
 * @author Pinger
 * @since 2018/1/11 0011 上午 11:28
 */

public class CustomViewActivity extends BaseActivity {

    @Override
    protected int getLayoutResID() {
        return R.layout.activity_custom_view;
    }

    @Override
    protected void initView() {
        setActionBar(getString(R.string.widget_custom), true);
    }


    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_falling:
                startActivity(FallingViewActivity.class);
                break;
            case R.id.btn_banner:
                startActivity(BannerViewActivity.class);
                break;
        }
    }
}

File: sample/src/main/java/com/leibo/repertory/widget/CustomViewActivity.kt
package com.leibo.repertory.widget

import android.view.View

import com.leibo.baselib.base.basic.BaseActivity
import com.leibo.repertory.R
import com.leibo.repertory.widget.banner.BannerViewActivity
import com.leibo.repertory.widget.falling.FallingViewActivity

/**
 * @author Pinger
 * @since 2018/1/11 0011 上午 11:28
 */

class CustomViewActivity : BaseActivity() {

    override val layoutResID: Int
        get() = R.layout.activity_custom_view

    override fun initView() {
        setActionBar(getString(R.string.widget_custom), true)
    }


    override fun onClick(view: View) {
        when (view.id) {
            R.id.btn_falling -> startActivity(FallingViewActivity::class.java)
            R.id.btn_banner -> startActivity(BannerViewActivity::class.java)
        }
    }
}
